chain/block.go

```diff
@@ -419,7 +419,7 @@ func (b *StatelessBlock) innerVerify(ctx context.Context, vctx VerifyContext) er
 	}
 
 	// Process transactions
-	results, ts, err := b.Execute(ctx, b.vm.Tracer(), parentView, feeManager, r)
+	results, ts, err := b.Execute(ctx, b.vm.ChainID(), b.vm.Tracer(), parentView, feeManager, r)
```

Comment:
Need to use the executing VM's `chainID` here to guard against replay attacks (signer signs the chainID they issue so this will guard against replay attacks across vm instances)